python/pyhail/dataset.py

```diff
@@ -956,10 +1050,43 @@ def filter_alleles(self, condition, annotation=None, subset=True, keep=True, fil
         return self.hc.run_command(self, pargs)
 
     def filter_genotypes(self, condition, keep=True):
-        """Filter variants based on expression.
+        """Filter genotypes based on expression.
+
+        **Examples**
+
+        Filter genotypes by allele balance dependent on genotype call
+
+        >>> (vds.filter_genotypes('let ab = g.ad[1] / g.ad.sum in'
+        >>>                      '((g.isHomRef && ab <= 0.1) ||'
+        >>>                      '(g.isHet && ab >= 0.25 && ab <= 0.75) ||'
+        >>>                      '(g.isHomVar && ab >= 0.9))'))
+
+        **Implementation Details**
+
+        Use the `Hail expression language <../reference.html#HailExpressionLanguage>`_ to supply a ``condition`` (boolean expression) involving the following exposed data structures:
+
+        - ``s`` (*Sample*): :ref:`sample`
+        - ``v`` (*Variant*): :ref:`variant`
+        - ``sa``: sample annotations
+        - ``va``: variant annotations
+        - ``global``: global annotations
+
+        All variables and values are case sensitive.
+
+        For more information, see the documentation on `data representation, annotations <../overview.html#>`_, and
+        the `expression language <../reference.html#HailExpressionLanguage>`_.
+
+        .. caution::
+
+            Missing values will always be **excluded**, regardless of whether ``keep=True`` or ``keep=False``.  Expressions in which any value is missing will evaluate to missing.
```

Comment:
This a bit unclear.  If the condition is missing, the genotype will always be excluded.